python/ray/ml/predictor.py

```diff
@@ -27,7 +28,27 @@ def from_checkpoint(cls, checkpoint: Checkpoint) -> "Predictor":
         """
         raise NotImplementedError
 
-    def predict(self, data: DataBatchType) -> DataBatchType:
+    def predict(
+        self, preprocessed_dataset: ray.data.Dataset, **iter_batches_kwargs
```

Comment:
hmm this does change the usage pattern in Serve. Should we consider building our own Dataset from DataBatchType for `predict`? cc @edoakes

Reply:
Can Serve just call `score_fn` directly and not call `predict`? So you would just feed in a batch of torch tensors. and not have to work with Datasets.

The names of functions can all be changed to match whatever you guys think make sense.

Reply:
Sure yeah, this was just copied from the prototype to get an end to end example working. We can change this back.

But why restrict `DataBatchType` to be a pandas Dataframe or numpy array? In this case, we want the batch to be a torch tensor.

Reply:
I think ideally Datasets can provide a batch format for torch and tf, so then for offline inference we can do something like

`ds.map_batches(lambda batch: predictor.predict(batch), batch_format="torch")`

Reply:
ok reverted back predictor back to what it was.